### PR TITLE
[FIX] mail: fix mention suggestion order and skip intermediate rpc

### PR DESCRIPTION
**Pull request by seb-odoo**
task-2387825

### CODE REVIEW

File: addons/mail/static/src/models/composer/composer.js [hunk 1]
```diff
@@ -566,6 +566,32 @@ function factory(dependencies) {
...
         }
 
+        /**
+         * Executes the given async function, only when the last function
+         * executed by this method terminates. If there is already a pending
+         * function it is replaced by the new one. This ensures the result of
+         * these function come in the same order as the call order, and it also
+         * allows to skip obsolete intermediate calls.
+         *
+         * @param {function} func
```
Comment (alexkuhn) [inline addons/mail/static/src/models/composer/composer.js:576]: missing `@private`

File: addons/mail/static/src/models/composer/composer.js [hunk 2]
```diff
@@ -566,6 +566,32 @@ function factory(dependencies) {
...
+            this._nextMentionRpcFunction = undefined;
+            try {
+                await func();
+            } finally {
+                this._hasMentionRpcInProgress = false;
+                if (this._nextMentionRpcFunction) {
+                    this._executeOrQueueFunction(this._nextMentionRpcFunction);
+                }
+            }
+        }
```
Comment (alexkuhn) [inline addons/mail/static/src/models/composer/composer.js:593]: missing `this.async()`

File: addons/mail/static/src/models/composer/composer.js [hunk 3]
```diff
@@ -566,6 +566,32 @@ function factory(dependencies) {
...
+         *
+         * @param {function} func
+         */
+        async _executeOrQueueFunction(func) {
+            if (this._hasMentionRpcInProgress) {
+                this._nextMentionRpcFunction = func;
+                return;
+            }
+            this._hasMentionRpcInProgress = true;
+            this._nextMentionRpcFunction = undefined;
```
Comment (alexkuhn) [inline addons/mail/static/src/models/composer/composer.js:584]: Would define them in `init()`. Easier to track special props of a class, which is super handful when refactoring.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Looks alright as a fix. Eventually will have RPC manager to solve this overall issue in Odoo.

File: addons/mail/static/src/models/composer/composer.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/composer/composer.js:576]: indeed

File: addons/mail/static/src/models/composer/composer.js (hunk 3 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/composer/composer.js:584]: I guess you mean `_willCreate`

Comment (alexkuhn) [reply addons/mail/static/src/models/composer/composer.js:584]: yes

File: addons/mail/static/src/models/composer/composer.js (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/composer/composer.js:593]: It probably doesn't matter at all in this case, especially with the  `finally`. But I guess it doesn't hurt to add it.

What might be better to add though is to actually check for exists before executing the function

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**